### ISSUE TITLE
Label displays, VBox does not

### ISSUE THREAD

**Issue by josePhoenix** [jupyter-widgets/ipywidgets#1450]
Potentially related to #1176. I can't seem to get a version of the widgets working that a) doesn't emit the warning about a version mismatch (even after nuking everything) and b) shows HBox and VBox contents. 

![screen shot 2017-06-22 at 2 51 46 pm](https://user-images.githubusercontent.com/172886/27450651-5c0e8b60-575a-11e7-8dad-730c53a3d0a9.png)

Using Python 3.6, ipywidgets 6.0, widgetsnbextension 2.0, notebook 5.0, Jupyter 4.3

**Comment by jasongrout** [comment]
This usually boils down to debugging confusion surrounding the notebook extension system.

(a) can you check to see if there are any errors in the Javascript console
(b) can you launch the notebook with `--debug`, which will print out a line showing where on disk it is finding the widgets javascript file. Make sure that is the same place you installed it.
(c) what does `jupyter nbextension list` say?

**Comment by josePhoenix** [comment]
Errors in the JS console:

```
[Error] Error

column: 27

error_stack: [Error: Class HBoxModel not found in module jupyter-js-widgets, Error: Couldn't create model, Error: Couldn't create a model.] (3)

line: 31924

message: "Couldn't create a model."

sourceURL: "http://localhost:8889/nbextensions/jupyter-js-widgets/extension.js?v=20170622162706"

stack: "WrappedError@http://localhost:8889/nbextensions/jupyter-js-widgets/extension.js?v=20170622162706:31924:2…"

Error Prototype

	promiseRejection (extension.js:32032)
	promiseReactionJob
[Error] Couldn't process kernel message
Error

column: 27

error_stack: [Error: Class HBoxModel not found in module jupyter-js-widgets, Error: Couldn't create model, Error: Couldn't create a model.] (3)

line: 31924

message: "Couldn't create a model."

sourceURL: "http://localhost:8889/nbextensions/jupyter-js-widgets/extension.js?v=20170622162706"

stack: "WrappedError@http://localhost:8889/nbextensions/jupyter-js-widgets/extension.js?v=20170622162706:31924:2…"

Error Prototype
	(anonymous function) (main.min.js:24481)
	promiseReactionJob
[Error] Couldn't process kernel message
Error

column: 27

error_stack: [Error: Class HBoxModel not found in module jupyter-js-widgets, Error: Couldn't create model, Error: Couldn't create a model.] (3)

line: 31924

message: "Couldn't create a model."

sourceURL: "http://localhost:8889/nbextensions/jupyter-js-widgets/extension.js?v=20170622162706"

stack: "WrappedError@http://localhost:8889/nbextensions/jupyter-js-widgets/extension.js?v=20170622162706:31924:2…"

Error Prototype
	(anonymous function) (main.min.js:24481)
	promiseReactionJob
```

Debug output:

```
$ jupyter notebook --debug
[D 16:28:24.965 NotebookApp] Searching ['/Users/jlong/software/wfirst-tools', '/Users/jlong/.jupyter', '/Users/jlong/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files
[D 16:28:24.966 NotebookApp] Looking for jupyter_config in /etc/jupyter
[D 16:28:24.966 NotebookApp] Looking for jupyter_config in /usr/local/etc/jupyter
[D 16:28:24.966 NotebookApp] Looking for jupyter_config in /Users/jlong/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/etc/jupyter
[D 16:28:24.966 NotebookApp] Looking for jupyter_config in /Users/jlong/.jupyter
[D 16:28:24.967 NotebookApp] Looking for jupyter_config in /Users/jlong/software/wfirst-tools
[D 16:28:24.967 NotebookApp] Looking for jupyter_notebook_config in /etc/jupyter
[D 16:28:24.967 NotebookApp] Looking for jupyter_notebook_config in /usr/local/etc/jupyter
[D 16:28:24.967 NotebookApp] Looking for jupyter_notebook_config in /Users/jlong/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/etc/jupyter
[D 16:28:24.968 NotebookApp] Looking for jupyter_notebook_config in /Users/jlong/.jupyter
[D 16:28:24.968 NotebookApp] Looking for jupyter_notebook_config in /Users/jlong/software/wfirst-tools
[I 16:28:25.301 NotebookApp] The port 8888 is already in use, trying another port.
[I 16:28:25.310 NotebookApp] Serving notebooks from local directory: /Users/jlong/software/wfirst-tools
[I 16:28:25.310 NotebookApp] 0 active kernels
[I 16:28:25.310 NotebookApp] The Jupyter Notebook is running at: xxx
[I 16:28:25.310 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 16:28:25.311 NotebookApp]

    Copy/paste this URL into your browser when you connect for the first time,
    to login with a token:
        xxx
[I 16:28:25.613 NotebookApp] Accepting one-time-token-authenticated connection from ::1
[D 16:28:25.614 NotebookApp] Using contents: services/contents
[D 16:28:25.644 NotebookApp] Path base/images/favicon.ico served from /Users/jlong/lib/python3.6/site-packages/notebook/static/base/images/favicon.ico
[D 16:28:25.645 NotebookApp] Path components/jquery-ui/themes/smoothness/jquery-ui.min.css served from /Users/jlong/lib/python3.6/site-packages/notebook/static/components/jquery-ui/themes/smoothness/jquery-ui.min.css
[D 16:28:25.645 NotebookApp] Path components/jquery-typeahead/dist/jquery.typeahead.min.css served from /Users/jlong/lib/python3.6/site-packages/notebook/static/components/jquery-typeahead/dist/jquery.typeahead.min.css
[D 16:28:25.645 NotebookApp] Path style/style.min.css served from /Users/jlong/lib/python3.6/site-packages/notebook/static/style/style.min.css
[D 16:28:25.646 NotebookApp] Path components/es6-promise/promise.min.js served from /Users/jlong/lib/python3.6/site-packages/notebook/static/components/es6-promise/promise.min.js
[D 16:28:25.646 NotebookApp] Path components/preact/index.js served from /Users/jlong/lib/python3.6/site-packages/notebook/static/components/preact/index.js
[D 16:28:25.647 NotebookApp] Path components/proptypes/index.js served from /Users/jlong/lib/python3.6/site-packages/notebook/static/components/proptypes/index.js
[D 16:28:25.647 NotebookApp] Path components/preact-compat/index.js served from /Users/jlong/lib/python3.6/site-packages/notebook/static/components/preact-compat/index.js
[D 16:28:25.647 NotebookApp] Path components/requirejs/require.js served from /Users/jlong/lib/python3.6/site-packages/notebook/static/components/requirejs/require.js
[D 16:28:25.648 NotebookApp] Path base/images/logo.png served from /Users/jlong/lib/python3.6/site-packages/notebook/static/base/images/logo.png
[D 16:28:25.648 NotebookApp] Path tree/js/main.min.js served from /Users/jlong/lib/python3.6/site-packages/notebook/static/tree/js/main.min.js
[D 16:28:25.650 NotebookApp] 200 GET /tree?token=6845f34d83fe7a32ca4325ee0ad6f4820846335eac254038 (::1) 37.62ms
[D 16:28:25.676 NotebookApp] Path custom.css served from /Users/jlong/lib/python3.6/site-packages/notebook/static/custom/custom.css
[D 16:28:25.677 NotebookApp] 304 GET /custom/custom.css (::1) 1.28ms
[D 16:28:25.783 NotebookApp] Path services/contents.js served from /Users/jlong/lib/python3.6/site-packages/notebook/static/services/contents.js
[D 16:28:25.784 NotebookApp] 200 GET /static/services/contents.js?v=20170622162824 (::1) 1.09ms
[D 16:28:25.809 NotebookApp] 200 GET /api/config/tree?_=1498163305732 (::1) 3.99ms
[D 16:28:25.813 NotebookApp] 200 GET /api/config/common?_=1498163305733 (::1) 2.38ms
[D 16:28:25.829 NotebookApp] 200 GET /api/terminals?_=1498163305734 (::1) 0.95ms
[D 16:28:25.832 NotebookApp] Found kernel python3 in /usr/local/share/jupyter/kernels
[D 16:28:25.832 NotebookApp] Found kernel python3 in /usr/local/share/jupyter/kernels
[D 16:28:25.835 NotebookApp] 304 GET /api/kernelspecs (::1) 4.25ms
[D 16:28:25.836 NotebookApp] 200 GET /api/terminals?_=1498163305736 (::1) 0.58ms
[D 16:28:25.838 NotebookApp] 200 GET /api/sessions?_=1498163305735 (::1) 1.65ms
[D 16:28:25.887 NotebookApp] 200 GET /api/contents?type=directory&_=1498163305737 (::1) 9.07ms
[D 16:28:25.951 NotebookApp] Path custom.js served from /Users/jlong/lib/python3.6/site-packages/notebook/static/custom/custom.js
[D 16:28:25.953 NotebookApp] 200 GET /custom/custom.js?v=20170622162824 (::1) 2.39ms
[D 16:29:17.056 NotebookApp] 200 GET /api/contents/Untitled1.ipynb?content=0&_=1498163235097 (::1) 1.37ms
[I 16:29:17.062 NotebookApp] Saving file at /Untitled1.ipynb
[D 16:29:17.062 NotebookApp] Saving /Users/jlong/software/wfirst-tools/Untitled1.ipynb
[D 16:29:17.082 NotebookApp] 200 PUT /api/contents/Untitled1.ipynb (::1) 20.72ms
```

```
$ jupyter nbextension list
Known nbextensions:
  config dir: /Users/jlong/.jupyter/nbconfig
    notebook section
      jupyter-js-widgets/extension  enabled
      - Validating: OK
```

**Comment by jasongrout** [comment]
Thanks. How did you install it widgetsnbextension? It looks like it is installed into your user config directory. Did you do that on purpose?

We recommend that you install widgetsnbextension and just enable it using `--sys-prefix`. Otherwise it gets confusing about where it is getting the javascript.

**Comment by josePhoenix** [comment]
I followed these installation instructions: https://ipywidgets.readthedocs.io/en/latest/user_install.html

**Comment by josePhoenix** [comment]
Tried to use `--sys-prefix`:

```
$ jupyter nbextension uninstall --py widgetsnbextension
Uninstalling jupyter-js-widgets jupyter-js-widgets/extension
Removing: /usr/local/share/jupyter/nbextensions/jupyter-js-widgets
```

```
$ jupyter nbextension enable --py widgetsnbextension --sys-prefix
Enabling notebook extension jupyter-js-widgets/extension...
      - Validating: problems found:
        - require?  X jupyter-js-widgets/extension
```

**Comment by jasongrout** [comment]
@josePhoenix - let's continue on gitter for a few minutes...

**Comment by josePhoenix** [comment]
The issue seems to have been a spurious `/usr/local/share/jupyter` folder that `jupyter nbextension` installs to. Since I use Homebrew Python, I was expecting that everything would go in `~/share`, so when I did my scorched-earth cleanout of all Jupyter-related files, I only searched $HOME.

`rm -rf /usr/local/share/jupyter && rm ~/.jupyter/nbconfig/notebook.json` followed by `pip3 uninstall widgetsnbextension ipywidgets`, then 

```
pip3 install ipywidgets
jupyter nbextension enable --py widgetsnbextension
```

sorted it all out.
